pkg/detector/detector.go

```diff
@@ -1330,32 +1330,33 @@ func (d *ResourceDetector) HandleClusterPropagationPolicyCreationOrUpdate(policy
 
 // CleanupResourceTemplateMarks removes marks, such as labels and annotations, from object referencing by objRef.
 func (d *ResourceDetector) CleanupResourceTemplateMarks(objRef workv1alpha2.ObjectReference, cleanupFunc func(obj metav1.Object)) error {
-	workload, err := helper.FetchResourceTemplate(d.DynamicClient, d.InformerManager, d.RESTMapper, objRef)
+	gvr, err := restmapper.GetGroupVersionResource(d.RESTMapper, schema.FromAPIVersionAndKind(objRef.APIVersion, objRef.Kind))
 	if err != nil {
-		// do nothing if resource template not exist, it might have been removed.
-		if apierrors.IsNotFound(err) {
-			return nil
-		}
-		klog.Errorf("Failed to fetch resource(kind=%s, %s/%s): %v", objRef.Kind, objRef.Namespace, objRef.Name, err)
+		klog.Errorf("Failed to convert GVR from GVK(%s/%s), err: %v", objRef.APIVersion, objRef.Kind, err)
 		return err
 	}
 
-	workload = workload.DeepCopy()
```

Comment:
Hi, all @zhzhuang-zju @whitewindmills, it seems we omit this deepcopy, do we don't need it anymore?

Reply:
yes, don't need it. we fetch directly workload from karmada-apiserver now, to reduce the chance of conflict.

Reply:
thanks for your response.